src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesWhenExpressions.java

```diff
@@ -0,0 +1,58 @@
+/*
+UnnecessaryParentheses
+tokens = (default)EXPR, IDENT, NUM_DOUBLE, NUM_FLOAT, NUM_INT, NUM_LONG, \
+         STRING_LITERAL, LITERAL_NULL, LITERAL_FALSE, LITERAL_TRUE, ASSIGN, \
+         BAND_ASSIGN, BOR_ASSIGN, BSR_ASSIGN, BXOR_ASSIGN, DIV_ASSIGN, \
+         MINUS_ASSIGN, MOD_ASSIGN, PLUS_ASSIGN, SL_ASSIGN, SR_ASSIGN, STAR_ASSIGN, \
+         LAMBDA, TEXT_BLOCK_LITERAL_BEGIN, LAND, LITERAL_INSTANCEOF, GT, LT, GE, \
+         LE, EQUAL, NOT_EQUAL, UNARY_MINUS, UNARY_PLUS, INC, DEC, LNOT, BNOT, \
+         POST_INC, POST_DEC
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
+
+public class InputUnnecessaryParenthesesWhenExpressions {
+
+    void test(Object o) {
+
+      switch (o) {
+          case Integer i when (i > 0) -> {}  // violation 'Unnecessary paren.* around expression'
+          // violation below 'Unnecessary paren.* around expression'
+          case String s when (!s.isEmpty()) -> {}
+
+          case Point(boolean x, boolean y, boolean z)
+              // violation below 'Unnecessary paren.* around expression'
+                    when (x && y) || z -> System.out.println();
+
+          case Point p when ((p.x && p.y) || p.z) -> System.out.println();
+          // 2 violations above:
+          //                   'Unnecessary paren.* around expression'
+          //                    'Unnecessary paren.* around expression'
```

Comment:
This line indentation is different than the line before.

Reply:
done all indentations are fixed